src/creatures/players/account/account.cpp

```diff
@@ -65,161 +72,162 @@ Account::Account(const std::string &email) : email_(email) {
-
-  if (db_tasks_ == nullptr) {
-      return ERROR_NULLPTR;
-  }
-  if (amount == 0)  {
-    return ERROR_NO;
-  }
-
-  uint32_t current_coins = 0;
-  this->GetCoins(&current_coins);
-  if ((current_coins + amount) < current_coins) {
-    return ERROR_VALUE_OVERFLOW;
-  }
-
-  std::ostringstream query;
-  query << "UPDATE `accounts` SET `coins` = " << (current_coins + amount)
-        << " WHERE `id` = " << id_;
+error_t Account::AddCoins(int32_t amount, CoinType_t coinType)
+{
+	std::string coins = "`coins`";
+	if (coinType == COIN_TYPE_DEFAULT || coinType == COIN_TYPE_TRANSFERABLE) {
+		coins = "`coins`";
+	} else if (coinType == COIN_TYPE_TOURNAMENT) {
+		coins = "`tournamentBalance`";
+	}
+	std::ostringstream query;
+	query << "UPDATE `accounts` SET " << coins << " = " << coins << " + " << amount << " WHERE `id` = " << id_;
 
-  db_tasks_->addTask(query.str());
-  return ERROR_NO;
+	db_tasks_->addTask(query.str());
+	return ERROR_NO;
 }
 
-error_t Account::RemoveCoins(uint32_t amount) {
-
-  if (db_tasks_ == nullptr) {
-      return ERROR_NULLPTR;
-  }
-
-  if (amount == 0)  {
-    return ERROR_NO;
-  }
-
-  uint32_t current_coins = 0;
-  this->GetCoins(&current_coins);
+error_t Account::RemoveCoins(int32_t amount, CoinType_t coinType) {
 
-  if ((current_coins - amount) > current_coins) {
-    return ERROR_VALUE_NOT_ENOUGH_COINS;
-  }
+	if (db_tasks_ == nullptr) {
+		return ERROR_NULLPTR;
+	}
 
-  std::ostringstream query;
-  query << "UPDATE `accounts` SET `coins` = "<< (current_coins - amount)
-        << " WHERE `id` = " << id_;
+	if (amount == 0)	{
+		return ERROR_NO;
+	}
 
-  db_tasks_->addTask(query.str());
+	std::string coins = "`coins`";
+	if (coinType == COIN_TYPE_DEFAULT || coinType == COIN_TYPE_TRANSFERABLE) {
+		coins = "`coins`";
+	} else if (coinType == COIN_TYPE_TOURNAMENT) {
+		coins = "`tournamentBalance`";
+	}
+	std::ostringstream query;
+	query << "UPDATE `accounts` SET " << coins << " = " << coins << " - " << amount << " WHERE `id` = " << id_;
 
-  return ERROR_NO;
+	db_tasks_->addTask(query.str());
+	return ERROR_NO;
 }
 
-error_t Account::RegisterCoinsTransaction(CoinTransactionType type,
-                                          uint32_t coins,
-                                          const std::string& description) {
-
-  if (db_ == nullptr) {
-      return ERROR_NULLPTR;
-  }
+error_t Account::RegisterCoinsTransaction(uint32_t time, uint8_t mode, uint32_t amount, uint8_t coinMode, std::string description, int32_t cust)
+{
+	if (db_ == nullptr) {
+		return ERROR_NULLPTR;
+	}
 
-  std::ostringstream query;
-  query << "INSERT INTO `coins_transactions` (`account_id`, `type`, `amount`,"
-          " `description`) VALUES (" << id_ << ", " << static_cast<uint16_t>(type) << ", "<< coins
-          << ", " << db_->escapeString(description) << ")";
+	std::ostringstream query;
+	query << "INSERT INTO `store_history` (`account_id`, `time`, `mode`, `amount`, `coinMode`, `description`, `cust`) VALUES ("
```

Comment:
Here the idea is to be generic to store all coins transactions, not only for the store, so if there are something new that use the coins, we don't need to create a new table